staging/src/k8s.io/kubectl/pkg/cmd/create/create_priorityclass.go

```diff
@@ -62,50 +92,110 @@ func NewCmdCreatePriorityClass(f cmdutil.Factory, ioStreams genericclioptions.IO
+	if err != nil {
+		return err
+	}
+
+	restConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	o.Client, err = schedulingclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	o.CreateAnnotation = cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag)
+
+	o.DryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+	discoveryClient, err := f.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+	o.DryRunVerifier = resource.NewDryRunVerifier(dynamicClient, discoveryClient)
+	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+	o.PrintObj = func(obj runtime.Object) error {
+		return printer.PrintObj(obj, o.Out)
+	}
+
+	return nil
+}
+
+// Run calls the CreateSubcommandOptions.Run in the PriorityClassOptions instance
+func (o *PriorityClassOptions) Run() error {
+	priorityClass, err := o.createPriorityClass()
 	if err != nil {
 		return err
 	}
 
-	var generator generate.StructuredGenerator
-	switch generatorName := cmdutil.GetFlagString(cmd, "generator"); generatorName {
-	case generateversioned.PriorityClassV1GeneratorName:
-		generator = &generateversioned.PriorityClassV1Generator{
-			Name:             name,
-			Value:            cmdutil.GetFlagInt32(cmd, "value"),
-			GlobalDefault:    cmdutil.GetFlagBool(cmd, "global-default"),
-			Description:      cmdutil.GetFlagString(cmd, "description"),
-			PreemptionPolicy: apiv1.PreemptionPolicy(cmdutil.GetFlagString(cmd, "preemption-policy")),
+	if err := util.CreateOrUpdateAnnotation(o.CreateAnnotation, priorityClass, scheme.DefaultJSONEncoder()); err != nil {
+		return err
+	}
+
+	if o.DryRunStrategy != cmdutil.DryRunClient {
+		createOptions := metav1.CreateOptions{}
+		if o.FieldManager != "" {
+			createOptions.FieldManager = o.FieldManager
+		}
+		if o.DryRunStrategy == cmdutil.DryRunServer {
+			if err := o.DryRunVerifier.HasSupport(priorityClass.GroupVersionKind()); err != nil {
+				return err
+			}
+			createOptions.DryRun = []string{metav1.DryRunAll}
+		}
+		var err error
+		priorityClass, err = o.Client.PriorityClasses().Create(context.TODO(), priorityClass, createOptions)
+		if err != nil {
+			return fmt.Errorf("failed to create clusterrolebinding: %v", err)
 		}
-	default:
-		return errUnsupportedGenerator(cmd, generatorName)
 	}
 
-	return o.CreateSubcommandOptions.Complete(f, cmd, args, generator)
+	return o.PrintObj(priorityClass)
 }
 
-// Run calls the CreateSubcommandOptions.Run in the PriorityClassOpts instance
-func (o *PriorityClassOpts) Run() error {
-	return o.CreateSubcommandOptions.Run()
+func (o *PriorityClassOptions) createPriorityClass() (*scheduling.PriorityClass, error) {
+	preemptionPolicy := corev1.PreemptionPolicy(o.PreemptionPolicy)
+	return &scheduling.PriorityClass{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
You're missing:
```go
		// this is ok because we know exactly how we want to be serialized
		TypeMeta: metav1.TypeMeta{APIVersion: scheduling.SchemeGroupVersion.String(), Kind: "PriorityClass"},
```
here

Reply:
done et rebased on master 🙂